codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -692,9 +665,185 @@ void decrementFlowControlWindow(int bytes) {
          * Discards this {@link FlowControlled}, writing an error. If this frame is in the pending queue,
          * the unwritten bytes are removed from this branch of the priority tree.
          */
-        void writeError(FlowControlled frame, Http2Exception cause) {
+        private void writeError(FlowControlled frame, Http2Exception cause) {
             decrementPendingBytes(frame.size());
             frame.error(cause);
         }
     }
+
+    /**
+     * The remote flow control state for a single stream that is not in a state where flow controlled frames cannot
+     * be exchanged.
+     */
+    private static final class ReducedFlowState extends RemoteFlowControlState {
```

Comment:
@Scottmitch I think the `ReducedFlowState` might not be such a good idea. 1) It adds complexity to the code 2) We will now always allocate a `ReducedFlowState` when a stream gets added, just to replace it by a real `FlowState` once it's active. My best guess would be that this "optimisation" will backfire, leading to more allocations and more memory in the common case.

If you think it's a good idea, could you explain a bit why you want to have this and do the math for some scenario showing that this makes sense?